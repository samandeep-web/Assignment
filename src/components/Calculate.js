import React from 'react';
import  './App.css';


const localeOptions = {
	style:'currency',
	currency:'CAD',
	minimumFractionDigits: 0,
    maximumFractionDigits: 0
}
export default function Calculate(props) {
	return(
	<div className="ui container"> 
   <div className="ui grey inverted segment">
     <div className="ui inverted form">
	  	<div className="ui grid">		
			<div className="row">	
			<div className="fourteen wide mobile four wide tablet four wide computer column fd">
			  <a className="ui tag label">Mortgage Amount</a></div>
				<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<input 
					type="number"
					id="mortgageAmount"  
					min="50000" 
					max="750000" 
					step="1000" 
					value={props.mortgageAmount} 
					onChange={(event) => 
						props.setMortgageAmount(event.target.value)
					} /></div>
			<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<span className="grid__item--header">{parseInt(props.mortgageAmount).toLocaleString('en-GB', localeOptions)}</span></div>
			</div></div><br/><br/>
			<div className="ui three row grid">
			<div className="row">	
			<div className="fourteen wide mobile four wide tablet four wide computer column fd">
			  <a className="ui tag label">Pre-Deposit Amount</a></div>
				<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<input 
					type="number"
					id="prePayment" 
					min="1000" 
					max="150000" 
					step="1000" 
					value={props.prePayment}
         			onChange={(event) =>
						props.setPrePayment(event.target.value)
         			} /></div>
        <div className="fourteen wide mobile four wide tablet four wide computer column fd">
			<span className="grid__item--header">{parseInt(props.prePayment).toLocaleString('en-GB', localeOptions)}</span></div>
			</div></div>
			<div className="ui three row grid">
			<div className="row">	
			<div className="fourteen wide mobile four wide tablet four wide computer column fd">
			  <a className="ui tag label">Mortgage Period</a></div>
				<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<input 
					type="range"
					className="grid__item--range-slider"
					id="mortgagePeriod"  
					min="5" 
					max="35" 
					step="1" 
					value={props.mortgagePeriod} 
					onChange={(event) => 
						props.setMortgagePeriod(event.target.value)
					} /></div>
				<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<span className="grid__item--header">{props.mortgagePeriod} Years</span>
			</div></div></div>
			<div className="ui three row grid">
			<div className="row">	
			<div className="fourteen wide mobile four wide tablet four wide computer column fd">
			  <a className="ui tag label">Interest Rate</a></div>
				<div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<input 
					type="range"
					className="grid__item--range-slider"
					id="interestRate"  
					min="0.1" 
					max="8" 
					step="0.1" 
					value={props.interestRate} 
					onChange={(event) => 
						props.setInterestRate(event.target.value)
					} /></div><div className="fourteen wide mobile four wide tablet four wide computer column fd">
				<span className="grid__item--header">{props.interestRate} %</span>
			</div></div></div>
		</div></div>
</div>	);
}