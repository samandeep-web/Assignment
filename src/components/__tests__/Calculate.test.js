import React from 'react';
import { shallow, mount } from 'enzyme';
import Calculate from '../Calculate';

// afterEach(() => {
//   wrapper.unmount();
//  });
describe('<Calculate/>', () => {
 const wrapper = shallow(<Calculate/>);
 it('should must match the snapshot', () => {
   expect(wrapper.html()).toMatchSnapshot();
 });


 it('should have mortgage amount field', () => {
  expect(wrapper.find('input[type="number"]').length).toBeGreaterThan(1);
 });

 it('should have proper props for mortgage amount field', () => {
  expect(wrapper.find('input[type="number"]').at(0).props()).toEqual({
    type:"number",
    id:"mortgageAmount", 
    min:"50000",
    max:"750000",
    step:"1000",
    onChange : expect.any(Function),
    value: undefined
  });
 });


 it('should have proper props for pre-deposit amount field', () => {
  expect(wrapper.find('input[type="number"]').at(1).props()).toEqual({
    type:"number",
    id:"prePayment",
    min:"1000",
    max:"150000",
    step:"1000",
    "value": undefined,
    "onChange": expect.any(Function)
  });
 });

 it('should have proper props for Mortgage period field', () => {
    expect(wrapper.find('input[type="range"]').at(0).props()).toEqual({
    type:"range",
    id:"mortgagePeriod",
    min:"5",
    max:"35",
    step:"1",
    "className": "grid__item--range-slider",
    "value": undefined,
    "onChange": expect.any(Function)		
  });
 });

 it('should have proper props Interest field', () => {
    expect(wrapper.find('input[type="range"]').at(1).props()).toEqual({
    type:"range",
    id:"interestRate",
    min:"0.1",
    max:"8",
    "className": "grid__item--range-slider",
    step:"0.1",
    "value": undefined,
    "onChange": expect.any(Function)		
  });
 });
});