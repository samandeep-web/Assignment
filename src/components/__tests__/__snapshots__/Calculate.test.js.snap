// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculate/> should must match the snapshot 1`] = `"<div class=\\"ui container\\"><div class=\\"ui grey inverted segment\\"><div class=\\"ui inverted form\\"><div class=\\"ui grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Mortgage Amount</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"number\\" id=\\"mortgageAmount\\" min=\\"50000\\" max=\\"750000\\" step=\\"1000\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">CA$NaN</span></div></div></div><br/><br/><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Pre-Deposit Amount</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"number\\" id=\\"prePayment\\" min=\\"1000\\" max=\\"150000\\" step=\\"1000\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">CA$NaN</span></div></div></div><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Mortgage Period</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"range\\" class=\\"grid__item--range-slider\\" id=\\"mortgagePeriod\\" min=\\"5\\" max=\\"35\\" step=\\"1\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\"> Years</span></div></div></div><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Interest Rate</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"range\\" class=\\"grid__item--range-slider\\" id=\\"interestRate\\" min=\\"0.1\\" max=\\"8\\" step=\\"0.1\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\"> %</span></div></div></div></div></div></div>"`;
